src/group/managed_group/mod.rs

```diff
@@ -123,60 +201,221 @@ impl ManagedGroup {
+                                // Since the Commit was applied without errors,
+                                // we can can call all corresponding callback
+                                // functions for the whole proposal list
+                                self.send_events();
+                                // We don't meed the pending proposals any longer
+                                self.pending_proposals = vec![];
+                            }
+                            Err(apply_commit_error) => match apply_commit_error {
+                                ApplyCommitError::SelfRemoved => {
+                                    self.active = false;
+                                }
+                                _ => {
+                                    self.invalid_message_event(InvalidMessageError::CommitError(
+                                        apply_commit_error,
+                                    ));
+                                }
+                            },
+                        }
+                    }
+                    MLSPlaintextContentType::Application(ref app_message) => {
+                        // If there is a callback for that event we should call it
+                        if let Some(app_message_received) =
+                            self.managed_group_config.callbacks.app_message_received
+                        {
+                            app_message_received(
+                                &self,
+                                aad_option.unwrap(),
+                                &plaintext.to_ref().sender,
+                                app_message,
+                            );
+                        }
+                    }
+                }
+            }
+        }
     }
 
-    // === Create application messages ===
+    // === Application messages ===
 
-    /// Create an application message
+    /// Creates an application message.  
+    /// Returns `ManagedGroupError::UseAfterEviction` if the member is no longer
+    /// part of the group. Returns `ManagedGroupError::
+    /// PendingProposalsExist` if pending proposals exist. In that case
+    /// `.flush_proposals()` must be called first and incoming messages from the
+    /// DS must be processed afterwards.
     pub fn create_message(
         &mut self,
         credential_bundle: &CredentialBundle,
         message: &[u8],
-    ) -> MLSMessage {
-        unimplemented!()
+    ) -> Result<MLSMessage, ManagedGroupError> {
+        if !self.active {
+            return Err(ManagedGroupError::UseAfterEviction);
+        }
+        if !self.pending_proposals.is_empty() {
+            return Err(ManagedGroupError::PendingProposalsExist);
+        }
+        let ciphertext =
+            self.group
+                .create_application_message(&self.aad, message, credential_bundle);
+        Ok(MLSMessage::Ciphertext(ciphertext))
+    }
+
+    /// Flush pending proposals
+    pub fn flush_proposals(
```

Comment:
Because here we just process the pending proposals that were already ratified by the DS. In the other functions you mention we add new proposals that were not yet sent to the DS.